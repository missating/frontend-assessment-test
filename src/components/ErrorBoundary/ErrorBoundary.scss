.errorboundary-container {
  display: flex;
  justify-content: center;
  width: 100%;
  margin: 2rem;

  &__content {
    background-color: #ffffff;
    width: 600px;
    padding: 10px;

    h1 {
      font-size: 3rem;
      margin: 0;
    }

    p {
      font-size: 0.7rem;
      font-style: oblique;
    }

    button {
      cursor: pointer;
      padding: 0.2rem 1rem;
      background: #fff;
      border: 1px solid var(--brand-color);
      font-size: 0.5rem;
      color: rgb(30, 29, 34);
      outline: none;
      transition: background 250ms ease-in-out, transform 150ms ease;
      -webkit-appearance: none;
      -moz-appearance: none;

      &:hover {
        background: var(--brand-color);
        color: #fff;
      }
    }
  }

}
