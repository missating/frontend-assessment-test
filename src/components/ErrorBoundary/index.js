// third party libraries
import React from 'react'
import PropTypes from 'prop-types';

// styles
import './ErrorBoundary.scss'

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  // eslint-disable-next-line no-unused-vars
  static getDerivedStateFromError(error) {
    // Update state so the next render will show the fallback UI.
    return { hasError: true };
  }

  render() {
    const {
      state: {
        hasError,
      },
      props: {
        children,
      },
    } = this;
    if (hasError) {
      return (
        <div className="errorboundary-container">
          <div className="errorboundary-container__content">
            <h1>500</h1>
            <p>Internal server error</p>
            <p>We are working on creating something better...</p>
            <button type="button" onClick={() => window.location.reload()}>Refresh</button>
          </div>

        </div>
      )
    }

    return children;
  }
}

export default ErrorBoundary

ErrorBoundary.propTypes = {
  children: PropTypes.node.isRequired,
}
