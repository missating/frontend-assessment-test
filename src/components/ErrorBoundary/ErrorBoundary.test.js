// third party libraries
import React from 'react'
import { mount } from 'enzyme';

// components
import ErrorBoundary from './index'

describe('ErrorBoundary Component', () => {
  const initialProps = {
    children: '<OtherComponent/>',
  }
  it('should be rendered properly', () => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const wrapper = mount(<ErrorBoundary {...initialProps} />);
    expect(wrapper).toMatchSnapshot();
  });
})
