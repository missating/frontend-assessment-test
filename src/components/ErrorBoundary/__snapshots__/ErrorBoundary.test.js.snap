// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary Component should be rendered properly 1`] = `
<ErrorBoundary>
  &lt;OtherComponent/&gt;
</ErrorBoundary>
`;
